main.diagnostics.tf

```diff
@@ -26,11 +26,11 @@ resource "azurerm_monitor_diagnostic_setting" "this" {
       category_group = enabled_log.value
     }
   }
-  dynamic "metric" {
+  dynamic "enabled_metric" {
```

Comment:
`metric` is deprecated and will be removed from the Azure provider for `enabled_metric`